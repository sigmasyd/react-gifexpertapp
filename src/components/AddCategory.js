import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setsInputValue] = useState('');

  const handleInputChange = (e) => {
    //console.log(e.target.value);
    setsInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('Submit hecho');
    if (inputValue.trim().length > 2) {
      setCategories((cat) => [inputValue, ...cat]);
      setsInputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
