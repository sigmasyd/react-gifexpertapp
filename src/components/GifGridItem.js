import React from 'react';

export const GifGridItem = ({ title, url }) => {
  //console.log(title, url);
  return (
    <div className="card animate__animated animate__fadeInLeft">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
