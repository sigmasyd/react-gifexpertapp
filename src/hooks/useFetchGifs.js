import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const [state, setstate] = useState({
    data: [],
    loading: true,
  });

  // se usa use effect para evitar que cada vez que un estado cambie... se renderize todo
  useEffect(() => {
    setTimeout(() => {
      getGifs(category).then((imgs) =>
        setstate({
          data: imgs,
          loading: false,
        })
      );
    }, 3000);
  }, [category]);
  return state;
};
