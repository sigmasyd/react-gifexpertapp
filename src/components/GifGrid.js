import React, { useEffect, useState } from 'react';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      'https://api.giphy.com/v1/gifs/search?api_key=anyoG6FgTEQ3Fx2KywvgvcZGe1b9Bv1Y&q=sakura&limit=10';
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });

    console.log(gifs);
    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      {images.map((img) => (
        <GifGridItem key={img.id} {...img} />
      ))}
    </>
  );
};
