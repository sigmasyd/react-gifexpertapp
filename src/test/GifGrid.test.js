import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../components/GifGrid';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs');

describe('Pruebas sobre el componente <GifGrid />', () => {
  const category = 'sakura';

  test('debe mostrar el componente correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://asdasdas',
        title: 'Cualquier cosas',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
});
