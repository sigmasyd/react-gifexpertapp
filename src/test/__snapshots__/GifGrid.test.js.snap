// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid /> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    sakura
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
